graphql-dxm-provider/src/main/java/org/jahia/modules/graphql/provider/dxm/admin/GqlClusterJournal.java

```diff
@@ -0,0 +1,116 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jahia.modules.graphql.provider.dxm.admin;
+
+import graphql.annotations.annotationTypes.GraphQLDescription;
+import graphql.annotations.annotationTypes.GraphQLField;
+import graphql.annotations.annotationTypes.GraphQLName;
+import org.apache.commons.lang.StringUtils;
+import org.hibernate.Session;
+import org.hibernate.SessionFactory;
+import org.hibernate.query.NativeQuery;
+import org.jahia.services.SpringContextSingleton;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+
+@GraphQLName("journal")
+@GraphQLDescription("Details about the Jahia cluster journal")
+public class GqlClusterJournal {
+
+    public static final Logger logger = LoggerFactory.getLogger(GqlClusterJournal.class);
+
+    // TODO (TECH-1284): Jahia core should provide a service to be able to get the necessary data instead of using hibernate and direct database queries
+    private final SessionFactory hibernateSessionFactory = (SessionFactory) SpringContextSingleton.getBean("sessionFactory");
+
+    @GraphQLField
+    @GraphQLName("globalRevision")
+    @GraphQLDescription("The latest revision of the journal on the cluster")
+    public Long getGlobalRevision() {
+        try (Session session = hibernateSessionFactory.openSession()) {
+            return queryGlobalRevision(session);
+        }
+    }
+
+    @GraphQLField
+    @GraphQLName("localRevision")
+    @GraphQLDescription("The latest revision of the journal on the local node")
+    public Long getLocalRevision() {
+        String currentNodeServerId = System.getProperty("cluster.node.serverId");
+        if (!StringUtils.isEmpty(currentNodeServerId)) {
+            try (Session session = hibernateSessionFactory.openSession()) {
+                return queryLocalRevision(session, currentNodeServerId);
+            }
+        } else {
+            logger.warn("Unable to query localRevision, cluster.node.serverId system property not found");
+            return null;
+        }
+    }
+
+    @GraphQLField
+    @GraphQLName("isClusterSync")
+    @GraphQLDescription("Is the journal in sync across all nodes of a cluster")
+    public Boolean isClusterSync() {
+        try (Session session = hibernateSessionFactory.openSession()) {
+            Long globalRevision = queryGlobalRevision(session);
+            if (globalRevision == null) {
+                throw new IllegalStateException("Unable to check if cluster is sync, globalRevision not found");
+            }
+
+            List<Long> revisions = new ArrayList<>(queryAllLocalRevisions(session));
+            revisions.add(globalRevision);
+
+            return revisions.stream().distinct().count() <= 1;
+        }
+    }
```

Comment:
This is nice to know if the cluster is sync or not
But how do I find which node is out of sync if all I can get is the local revision.

I think we should be able to return journal_id/revision number to have a better view of the cluster status (there maybe still some old cluster_id registered for example (saw that before))